pkg/frontend/transport/handler.go

```diff
@@ -277,7 +282,15 @@ func (f *Handler) reportQueryStats(r *http.Request, queryString url.Values, quer
 		"sharded_queries", stats.LoadShardedQueries(),
 		"split_queries", stats.LoadSplitQueries(),
 		"estimated_series_count", stats.GetEstimatedSeriesCount(),
-	}, formatQueryString(queryString)...)
+	}, formatQueryString(details, queryString)...)
+
+	if details != nil {
+		logMessage = append(logMessage,
+			"length", details.End.Sub(details.Start).String(),
+			"time_since_param_start", time.Since(details.Start).String(),
+			"time_since_param_end", time.Since(details.End).String(),
```

Comment:
This block is code assumes there's always start/end, but we have query APIs without that (e.g. /metadata). It could be misleading logging these values for API endpoints where start/end has no meaning.

Reply:
I agree about the /metadata requests. I removed the log lines in case the times are unset.

But I'm not sure whether having different log fields for `time_since*` will be helpful. The point of these is to ease spotting changes in query patterns. The loki queries will become more complicated if we have to account for multiple fields. I think changing them to instead `time_to_min_time` and `time_to_max_time` will make more sense.

This requires the changes you suggested in the other comment for min/max time. In the meantime I can remove the `time_since_param_end` and `time_since_param_start` log fields. 


WDYT?

Reply:
i added `time_since_max_time` and `time_since_min_time` and `length` all based on the min/max times of the query instead of start/end.